### PR TITLE
Adding some basic search capabilities

### PR DESCRIPTION
**Pull request by bmartus**
I needed a way to see if a certain type of product was in the users cart. This allows you to send an array of key-val pairs and it will return an array of items where all options match.

Let me know if you can think of anything else to add or test, I don't mind updating.

### CODE REVIEW

**Comment by lukepolo** [comment]
Awesome only suggestion is to return the items that match as well ? Thoughts ?

**Comment by bmartus** [comment]
The returned array contains all matching cart items.  Or did you mean inspect those  in the tests, rather than just count the size of the array?

**Comment by lukepolo** [comment]
Yah for the test just make sure the item when returned actually has those options

**Comment by bmartus** [comment]
Will do.

File: src/CartItem.php [hunk 1]
```diff
@@ -92,6 +92,23 @@ public function getHash()
...
     /**
+     * Search for matching options on the item
+     *
+     * @return mixed
+     */
+    public function find($data)
+    {
+        foreach ($data as $key => $value) {
+            if ($this->$key !== $value) {
+                // all search criteria must match
```
Comment (lukepolo) [inline src/CartItem.php]: remove comment , no need

**Comment by bmartus** [comment]
Let me know if you think any other tests would help.

**Comment by lukepolo** [comment]
I just made a huge update to laracart. Sadly this caused conflicts with yours.

**Comment by lukepolo** [comment]
ill look at it in a bit to merge.

**Comment by bmartus** [comment]
I can move these changes into a new PR off master tonight if that's easier, too.

**Comment by lukepolo** [comment]
That would be great.

**Comment by bmartus** [comment]
got it.

**Pull request closed without merging**